### PR TITLE
🚧⚡ Upgrade to webpacker 6 (Shakapacker)

### PR DESCRIPTION
**Pull request by josemigallas**
[THREESCALE-9449: Upgrade to webpacker 6 (shakapacker)](https://issues.redhat.com/browse/THREESCALE-9449)

Find more info about Shakapacker here https://github.com/shakacode/shakapacker/tree/6-stable

This is finally the upgrade we were looking for since... forever. A summary of important changes:
- `postcss` is now optional so it's been removed altogether.
- `webpacker` is now called `shakapacker` and in future releases even internal names will be renamed.
- Styles were failing either due to wrong import order or to missing stylesheets, therefore we're importing now the whole patternfly.css which is how it's supposed to be done (optimization will come later, maybe).
- packs are now queued using `append_javascript_pack_tag` and added to the DOM all at once with `javascript_pack_tag` at the layout level. Some critical points such as dev portal where we don't have full control will not use the `append` strategy.
- css is also split in chunks now and is needs to be added manually using `stylesheet_pack_tag`.
- javascript packs are now _deferred_ by default, that is they are loaded in a separate threat and do not block the initial HTML/CSS rendering. Scripts that inject functions into the global scope (window) still need to be loaded first.

Some TODOs left:
- quickstarts are broken after the upgrade, the package is super old and something must be not anymore compatible. Since it has never been approved for production I think we can simply disable it and fix it afterwards.
- `react/no-object-type-as-default-prop` is a new eslint rule that collides with what we have, but it looks kinda important so it's been disabled now but will be fixed afterwards.

### CODE REVIEW

File: features/step_definitions/asset_host_steps.rb [hunk 1]
```diff
@@ -6,20 +6,16 @@
   Rails.configuration.three_scale.asset_host = cdn_url
 end
 
-Then /^((javascript|font)\s)?assets should be loaded from the asset host$/ do |asset_type|
+Then "(javascript )assets should be loaded from the asset host" do
```
Comment (mayorova) [inline features/step_definitions/asset_host_steps.rb:9]: Why remove font loading test?

**Comment by josemigallas** [comment]
Closed in favor of #3357

File: features/step_definitions/asset_host_steps.rb (hunk 1 above)
Comment (josemigallas) [reply features/step_definitions/asset_host_steps.rb:12]: Right now css is being "injected" in the DOM and this test asserts the font-family prop has the correct url. But after this upgrade, stylesheets are linked and downloaded as a separate source:
```html
<link rel="stylesheet" media="screen" href="http://provider-admin.3scale.localhost:3000/packs/css/provider.css">
```
So this assertion is not possible anymore.

Comment (akostadinov) [reply features/step_definitions/asset_host_steps.rb:12]: Shouldn't they still be downloaded from CDN?

Comment (josemigallas) [reply features/step_definitions/asset_host_steps.rb:12]: They are. But this test does not test that. It tests that this is present **in the DOM**:
```css
@font-face {
    font-family: "RedHatDisplay";
    src: url(/packs/static/@patternfly/patternfly/assets/fonts/RedHatDisplay/RedHatDisplay-Regular-0c3b86386764101597ac.eot);
    src: url(/packs/static/@patternfly/patternfly/assets/fonts/RedHatDisplay/RedHatDisplay-Regular-0c3b86386764101597ac.eot?#iefix) format("embedded-opentype"), url(/packs/static/@patternfly/patternfly/assets/fonts/RedHatDisplay/RedHatDisplay-Regular-7791e86cb49866c2b138.woff) format("woff");
...
```
Which isn't anymore. Now this css is downloaded as an external file (like js)

**Pull request closed without merging**